src/constraint/dart/InverseKinematicsSampleable.cpp

```diff
@@ -221,7 +221,7 @@ bool IkSampleGenerator::sample(statespace::StateSpace::State* _state)
     mInverseKinematics->getTarget()->setTransform(poseState.getIsometry());
 
     // Run the IK solver. If it succeeds, return the solution.
-    if (mInverseKinematics->solve(true))
+    if (mInverseKinematics->solveAndApply(false))
```

Comment:
The behavior with this change will be different from the original behavior. What is the expected behavior here?

Reply:
Wait, @jslee02 , so in the original behavior, even incomplete solutions were applied to the skeleton?

Reply:
Yes, that's the motivation of the new APIs to be more clear on the behavior.